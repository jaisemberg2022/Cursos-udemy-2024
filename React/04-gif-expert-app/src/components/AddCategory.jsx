import TextField from "@mui/material/TextField";
import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState();
  const onInputChange = ({target})=>{
    setInputValue(target.value)
  }
  const onSubmit = (event)=>{
    event.preventDefault();
    if (inputValue.trim().lenght <=1 ) {return}
    // setCategories( (categories) =>[inputValue,...categories]);
    onNewCategory(inputValue.trim())
    setInputValue('');
  }
  return (
    <form onSubmit={onSubmit}>
        <TextField
          value={inputValue}
          id="outlined-basic"
          label="filtrar"
          variant="outlined"
          placeholder="Buscar gifs..."
          InputLabelProps={{ shrink: true }}
          onChange={onInputChange}
          autoComplete="off"
          size="small"
        />
    </form>
  );
};
