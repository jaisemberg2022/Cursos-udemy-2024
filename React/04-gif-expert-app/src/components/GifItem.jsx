import React from "react";
import { Divider, Card, CardContent, CardMedia } from "@mui/material";
import PropTypes from 'prop-types'

export const GifItem = ({ url,title }) => {
  return (
    <>
      <Card sx={{flex:1,height:'auto'}}>
        <CardMedia component="img" alt={title} image={url} />
        <CardContent>{title}</CardContent>
      </Card>
      <Divider />
    </>
  );
};

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired

}
