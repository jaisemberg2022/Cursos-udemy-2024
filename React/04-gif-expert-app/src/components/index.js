export * from './AddCategory';
export * from './GifExpert';
export * from './GifGrid';
export * from './GifItem';

