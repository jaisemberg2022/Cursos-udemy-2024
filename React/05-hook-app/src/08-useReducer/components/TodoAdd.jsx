import React, { useState } from "react";
import { Button, Stack, TextField } from "@mui/material";
import {useForm} from '../../hooks'


const TodoAdd = ({onNewTodo}) => {
  

  const {description,onInputChnge,onDeletForm} = useForm({
    description:''
  })

  const onSubmit =(event)=>{
    event.preventDefault();
    if (description.length <=1) return;
    const newTodo = {
      id:new Date().getDate() * 3,
      donde:false,
      description
    }
    onNewTodo(newTodo);
    onDeletForm()
  }


  return (
    <Stack
    gap={1}
    component={"form"}
    display={"inline-flex"}
    flexDirection={"column"}
    alignItems={"start"}
    onSubmit={onSubmit}
  >
      <TextField
        size="small"
        InputLabelProps={{ shrink: true }}
        label={"accion"}
        placeholder="que hay qye hacer?"
        autoComplete="off"
        value={description || ""}
        name="description"
        onChange={onInputChnge}
      />
      <Button type="submit" variant="contained" disableElevation size="small">
        enviar
      </Button>
    </Stack>
  );
};

export default TodoAdd;
