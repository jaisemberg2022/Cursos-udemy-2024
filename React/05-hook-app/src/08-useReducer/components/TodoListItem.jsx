import React from 'react'
import { Button,Divider,ListItem,ListItemText } from '@mui/material'


const TodoListItem = ({id,description,onDeleteTodo}) => {
  return (
    <>
        <ListItem  >
            <ListItemText primary={description} />
            <Button size='small' variant='contained' disableElevation color='error' onClick={()=>onDeleteTodo(id)}>Borrar</Button>
        </ListItem>
        <Divider/>
    </>
  )
}

export default TodoListItem