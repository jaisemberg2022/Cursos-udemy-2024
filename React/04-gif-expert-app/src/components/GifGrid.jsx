import React, { useEffect, useState } from "react";
import { ListItem, ListItemText, Divider } from "@mui/material";
import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";


export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);
  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };
  useEffect(() => {
    getImages();
  }, [category]);

  return (
    <>
      {images.map(({ id, title, url }) => (
        <GifItem key={id}/>
      ))}
    </>
  );
};
