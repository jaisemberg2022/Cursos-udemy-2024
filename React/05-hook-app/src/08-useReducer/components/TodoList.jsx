import React from 'react'
import { List } from '@mui/material'
import TodoListItem from './TodoListItem'


const TodoList = ({todos,onDeleteTodo}) => {
  return (
    <>
        <List  dense={true} sx={{width:'50%',flexBasis:'50%',backgroundColor:'white'}}>
            {
                todos.map(todo => (
                    <TodoListItem key={todo.id} id={todo.id} description={todo.description} onDeleteTodo={onDeleteTodo}/>
                ))
            }
        </List>
    </>
  )
}

export default TodoList