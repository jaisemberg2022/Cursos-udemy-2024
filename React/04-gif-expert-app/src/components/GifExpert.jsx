import React, { useState } from "react";
import {Typography,Stack,Box,List} from "@mui/material";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

export const GifExpert = () => {
  const [categories, setcategories] = useState(["Dragon ball"]);
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setcategories([newCategory, ...categories]);
  };
  return (
    <>
      <Box display={'inline-flex'} flexDirection={'column'} p={3} gap={2}>
        {/* titulo */}
        <Typography variant="h2">Titulo</Typography>
        {/* input */}
        <AddCategory onNewCategory={onAddCategory}/>
        {/* imagenes */}
        <Stack display={"inline-flex"} alignItems={"flex-start"} gap={2} boxShadow={2} bgcolor={"white"}>
          {/* <List sx={{flexBasis:'100%', width:'100%'}} > */}
            {categories.map((categorie) => (<GifGrid key={categorie}  category={categorie}/>))}
          {/* </List> */}
        </Stack>
      </Box>
    </>
  );
};
