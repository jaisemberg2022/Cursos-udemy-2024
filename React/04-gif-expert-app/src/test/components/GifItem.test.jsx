import {render,screen}  from '@testing-library/react'
import {GifItem} from '../../components/GifItem' 

describe('pruabas sobre <GifItem/>', () => { 

    const title = 'Titulo';
    const url = 'htpps://any.jpg';

    test('Debe hacer match con el snapshot',()=>{
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el "url" y el "atl" indicado',()=>{
        render(
            <GifItem title={title} url={url}/>
        );
        const {src , alt} = screen.getByRole('img');
        expect(alt).toBe(title);
        expect(src).toBe(url);
        
    });

    test('debe mostrar el titulo en el componente', () => { 
        render(
            <GifItem title={title} url={url}/>
        );
        expect(screen.getByText(title)).toBeTruthy();
     })
 })