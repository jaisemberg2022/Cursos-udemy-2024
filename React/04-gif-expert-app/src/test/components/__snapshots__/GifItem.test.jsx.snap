// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruabas sobre <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-9yf9xd-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Titulo"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      src="htpps://any.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      Titulo
    </div>
  </div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
</div>
`;
